### PR TITLE
rec: Structured logging for webserver

### PR DESCRIPTION
**Pull request by omoerbeek**
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Review by fredmorcos — APPROVED** [review]
Minor nitpicks, nothing blocking.

File: pdns/webserver.cc [hunk 1]
```diff
@@ -278,34 +291,41 @@ void WebServer::handleRequest(HttpRequest& req, HttpResponse& resp) const
...
+      SLOG(g_log<<Logger::Debug<<req.logprefix<<"No route found for \"" << req.url.path << "\"" << endl,
+           log->info(Logr::Debug, "No route found"));
       throw HttpNotFoundException();
     }
 
+    const string msg = "HTTP ISE Exception";
     try {
       handler(&req, &resp);
-      g_log<<Logger::Debug<<req.logprefix<<"Result for \"" << req.url.path << "\": " << resp.status << ", body length: " << resp.body.size() << endl;
+      SLOG(g_log<<Logger::Debug<<req.logprefix<<"Result for \"" << req.url.path << "\": " << resp.status << ", body length: " << resp.body.size() << endl,
```
[the remark this answers is not in the document]
Comment (omoerbeek) [reply pdns/webserver.cc:302]: I don't want to change the old style logging, unless I encounter a real bug

File: pdns/webserver.cc [hunk 2]
```diff
@@ -179,7 +182,8 @@ void WebServer::apiWrapper(const WebServer::HandlerFunction& handler, HttpReques
   }
 
   if (!auth_ok) {
-    g_log<<Logger::Error<<req->logprefix<<"HTTP Request \"" << req->url.path << "\": Authentication by API Key failed" << endl;
+    SLOG(g_log<<Logger::Error<<req->logprefix<<"HTTP Request \"" << req->url.path << "\": Authentication by API Key failed" << endl,
+         d_slog->info(Logr::Error, "HTTP Request: Authentication by API Key failed", "urlpath", Logging::Loggable(req->url.path)));
```
[the remark this answers is not in the document]
Comment (omoerbeek) [reply pdns/webserver.cc:186]: Hard to decide... I'll leave it for now.

**Review by chbruyand — APPROVED** [review]
LGTM!

**Pull request merged**